nomenclature/codelist.py

```diff
@@ -212,7 +212,7 @@ def from_directory(cls, name: str, path: Path, file_glob_pattern: str = "**/*"):
                 code.file = yaml_file.relative_to(path.parent).as_posix()
                 code_list.append(code)
         code_list = cls._parse_and_replace_tags(code_list, path, file_glob_pattern)
-        mapping: Dict[str, Code] = {}
+        mapping: Dict[str, cls.code_basis] = {}
```

Comment:
Same argument as above
```suggestion
        mapping: Dict[str, Code] = {}
```